ops/cli.py

```diff
@@ -1,351 +1,343 @@
 #!/usr/bin/env python3
-"""WFL Deployment Script
-@rex
+"""
+WFL Deployment Script
 
-requirements: pip3 install gitpython pyyaml
+requirements: pip3 install pyyaml
 
 usage: python3 cli.py -h
 """
-from enum import Enum
-from pathlib import Path
-from typing import Callable
 import argparse
 import json
 import os
 import shutil
 import subprocess
 import sys
-import tempfile
+from typing import Callable
+
 import yaml
 
+from render_ctmpl import render_ctmpl
+from util.misc import info, success, error, warn, shell
+
 # In case this will need to run on Windows systems
 if sys.platform.lower() == "win32":
     os.system("color")
 
-class AvailableColors(Enum):
-    GRAY = 90
-    RED = 91
-    GREEN = 92
-    YELLOW = 93
-    BLUE = 94
-    PURPLE = 95
-    WHITE = 97
-    BLACK = 30
-    DEFAULT = 39
+
+class Config:
+    """Base class for storing/resolving/validating command line input."""
+
+    def __init__(self, parsed_args: argparse.Namespace):
```

Comment:
No other `Config` seems to override this. This whole class can be replaced with a `Dictionary` returned by a `parse_input` function, whose keys were the field names of this class. Then:

1. You no longer need to override `__init__`
2. You no longer need inheritance - can use strategy pattern instead if this was Java, BUT
3. You no longer need the command classes because functions are objects in python

Reply:
As a final thought - this is a deployment script. Does it really need to be this engineered?

Reply:
I see very very little benefit to using a dictionary here. If we want to go that route, that's okay, but we need to be aware that in doing so we are making the script less checked.

My entire thinking about deployment scripts is this: when it is 2am and one of us is awake because our prod went down, a fully-fledged script with helpful error text, argument validation, and sane but overridable defaults is what I want to be working with. Engineering it in this way means that our tooling can help check our work so we can catch mistakes earlier and fix them quicker. Production isn't going to break because of this script, but this script is what will help fix it, and I think it is worth it to put the time in to prepare this script for that possibility.

Reply:
Resolved in new version. No methods exist in the file, no inheritance exists in the file, commands are not classes. "Commands" are now done via a higher-order function such that individual steps can be very easily commented out.